wgpu-types/src/lib.rs

```diff
@@ -2109,6 +2037,467 @@ pub enum TextureFormat {
+            }
+            TextureFormat::Bc7RgbaUnormSrgb => {
+                let name = "bc7-rgba-unorm-srgb";
+                serializer.serialize_unit_variant(name, 60, name)
+            }
+            TextureFormat::Etc2Rgb8Unorm => {
+                let name = "etc2-rgb8unorm";
+                serializer.serialize_unit_variant(name, 61, name)
+            }
+            TextureFormat::Etc2Rgb8UnormSrgb => {
+                let name = "etc2-rgb8unorm-srgb";
+                serializer.serialize_unit_variant(name, 62, name)
+            }
+            TextureFormat::Etc2Rgb8A1Unorm => {
+                let name = "etc2-rgb8a1unorm";
+                serializer.serialize_unit_variant(name, 63, name)
+            }
+            TextureFormat::Etc2Rgb8A1UnormSrgb => {
+                let name = "etc2-rgb8a1unorm-srgb";
+                serializer.serialize_unit_variant(name, 64, name)
+            }
+            TextureFormat::Etc2Rgba8Unorm => {
+                let name = "etc2-rgba8unorm";
+                serializer.serialize_unit_variant(name, 65, name)
+            }
+            TextureFormat::Etc2Rgba8UnormSrgb => {
+                let name = "etc2-rgba8unorm-srgb";
+                serializer.serialize_unit_variant(name, 66, name)
+            }
+            TextureFormat::EacR11Unorm => {
+                let name = "eac-r11unorm";
+                serializer.serialize_unit_variant(name, 67, name)
+            }
+            TextureFormat::EacR11Snorm => {
+                let name = "eac-r11snorm";
+                serializer.serialize_unit_variant(name, 68, name)
+            }
+            TextureFormat::EacRg11Unorm => {
+                let name = "eac-rg11unorm";
+                serializer.serialize_unit_variant(name, 69, name)
+            }
+            TextureFormat::EacRg11Snorm => {
+                let name = "eac-rg11snorm";
+                serializer.serialize_unit_variant(name, 70, name)
+            }
+            TextureFormat::Astc { block, channel } => {
+                let block = match block {
+                    AstcBlock::B4x4 => "4x4",
+                    AstcBlock::B5x4 => "5x4",
+                    AstcBlock::B5x5 => "5x5",
+                    AstcBlock::B6x5 => "6x5",
+                    AstcBlock::B6x6 => "6x6",
+                    AstcBlock::B8x5 => "8x5",
+                    AstcBlock::B8x6 => "8x6",
+                    AstcBlock::B8x8 => "8x8",
+                    AstcBlock::B10x5 => "10x5",
+                    AstcBlock::B10x6 => "10x6",
+                    AstcBlock::B10x8 => "10x8",
+                    AstcBlock::B10x10 => "10x10",
+                    AstcBlock::B12x10 => "12x10",
+                    AstcBlock::B12x12 => "12x12",
+                };
+
+                let channel = match channel {
+                    AstcChannel::Unorm => "unorm",
+                    AstcChannel::UnormSrgb => "unorm-srgb",
+                    AstcChannel::Hdr => "hdr",
+                };
```

Comment:
I was going to call `AstcChannel`'s ser/de impl, but unsure if thats even possible